<launch>

  <!-- Camera -->
  <include file="$(find fiducial_detect)/launch/pgr_camera.launch"/>
  
  <!-- Undistort -->
<!--
  <node name="undistort" pkg="image_proc" type="image_proc" 
        ns="/pgr_camera_node" respawn="true"/>
-->
  
  <!-- Camera transform -->
  <include file="$(find fiducial_detect)/launch/pgr_pose.launch"/>
  
  <!-- Fiducial detection -->
  <include file="$(find fiducial_detect)/launch/fiducials_no_odom.launch">
    <arg name="camera" value="/pgr_camera_node"/>
    <arg name="image" value="image_raw"/>
    <param name="fiducial_len" type="double" value="0.145"/>
    <param name="undistort_points" type="bool" value="true"/>
  </include>

  <!-- Fiducial slam -->
  <node type="fiducial_slam.py" pkg="fiducial_slam" output="screen" name="fiducials_slam">
    <param name="map_file" value="$(env HOME)/.ros/slam/map.txt" />
    <param name="obs_file" value="$(env HOME)/.ros/slam/obs.txt" />
    <param name="trans_file" value="$(env HOME)/.ros/slam/trans.txt" />
    <param name="publish_tf" value="false" />
    <param name="camera_frame" value="pgr_camera_frame" />
    <remap to="/fiducials_localization/fiducial_transforms" from="/fiducial_transforms"/>
  </node>
</launch>
